### PR TITLE
prepare release 0.30.0

### LINKED ISSUE #1489: when will the 0.29.1 or the 0.30.0 be released?

### PR DESCRIPTION
**Pull request by tyler-french**
Following the https://github.com/bazelbuild/bazel-gazelle/wiki/Release-process, this PR prepares the 0.30.0 release.

**What type of PR is this?**

Release preparation

**What package or component does this PR mostly affect?**

all

**What does this PR do? Why is it needed?**

Prepare the 0.30.0 release

**Which issues(s) does this PR fix?**

Fixes #1489

**Other notes for review**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -151,19 +151,19 @@ should look like this:
...
-            "https://mirror.bazel.build/github.com/bazelbuild/rules_go/releases/download/v0.37.0/rules_go-v0.37.0.zip",
-            "https://github.com/bazelbuild/rules_go/releases/download/v0.37.0/rules_go-v0.37.0.zip",
+            "https://mirror.bazel.build/github.com/bazelbuild/rules_go/releases/download/v0.38.1/rules_go-v0.38.1.zip",
+            "https://github.com/bazelbuild/rules_go/releases/download/v0.38.1/rules_go-v0.38.1.zip",
         ],
     )
 
     http_archive(
         name = "bazel_gazelle",
         sha256 = "ecba0f04f96b4960a5b250c8e8eeec42281035970aa8852dda73098274d14a1d",
```
Comment (tyler-french) [inline README.rst:163]: Not sure how to update this SHA when updating the SHA will change the SHA :)

Comment (malt3) [reply README.rst:163]: Looking at the last release, the tagged commit still shows old versions / hashes in the README: https://github.com/bazelbuild/bazel-gazelle/tree/v0.29.0

The versions were updated afterwards: #1419

**Comment by andyliuliming** [comment]
#1489

**Comment by andyliuliming** [comment]
@tyler-french https://github.com/bazelbuild/rules_go/releases/tag/v0.39.0 just go released :)

**Review by fmeum — APPROVED** [review]

**Pull request merged**